src/common/vulkan_platform.h

```diff
@@ -32,9 +32,21 @@
 // redefined to be nullptr). This keeps the type-safety of having the handles be different types
 // (like vulkan.h on 64 bit) but makes sure the types are different on 32 bit architectures.
 
+// Force the handle type to have the same alignment as what would have been the Vulkan
+// non-dispatchable handle type.
+#if defined(DAWN_PLATFORM_64_BIT)
+// In 64 bit handles are pointers to some structure, we just declare one inline here.
+using NativeVulkanHandleType = struct VkSomeHandle*;
+#elif defined(DAWN_PLATFORM_32_BIT)
+using NativeVulkanHandleType = uint64_t;
```

Comment:
Maybe a dumb question, but why not just use uint64_t in all cases? In 64-bit, we know a pointer is 64-bit, in 32-bit we're forcing it to 64 bits.

Reply:
It's because for type-safety the Vulkan headers use `uint64_t` on 32bit but `struct VkType_T*` on 64bit that can't cast to one another. So we have to do the same in case the size and alignment of pointers isn't the same on 64bit.